packages/opentelemetry-exporter-prometheus/src/PrometheusExporter.ts

```diff
@@ -26,16 +26,17 @@ import * as url from 'url';
 import { ExporterConfig } from './export/types';
 import { PrometheusSerializer } from './PrometheusSerializer';
 import { PrometheusLabelsBatcher } from './PrometheusLabelsBatcher';
-
 export class PrometheusExporter implements MetricExporter {
   static readonly DEFAULT_OPTIONS = {
-    port: 9464,
+    host: process.env.OTEL_EXPORTER_PROMETHEUS_HOST ?? '0.0.0.0',
+    port: Number(process.env.OTEL_EXPORTER_PROMETHEUS_PORT) ?? 9464,
```

Comment:
Not sure if `Number` constructor provides good results in all cases? Maybe use `parseInt(val, 10)` and check for `NaN` instead.

Reply:
The internal mechanism (`http.server.listen()`) will throw for invalid ports that become `NaN` or are outside of the range:

```console
> server.listen({port: NaN})
Thrown:
RangeError [ERR_SOCKET_BAD_PORT]: Port should be >= 0 and < 65536. Received NaN.
```

I think surfacing the error is better than falling back to 9464 in this case, do you think adding an error handler like:

```js
      this.startServer().then(callback).catch(err => this._logger.error(err));
```

to log these would be good?